examples/jeventdemo/source/src/org/kaaproject/kaa/demo/event/EventDemo.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.event;
+
+import java.io.IOException;
+import java.lang.String;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.kaaproject.kaa.client.DesktopKaaPlatformContext;
+import org.kaaproject.kaa.client.Kaa;
+import org.kaaproject.kaa.client.KaaClient;
+import org.kaaproject.kaa.client.SimpleKaaClientStateListener;
+import org.kaaproject.kaa.client.event.EventFamilyFactory;
+import org.kaaproject.kaa.client.event.EventListenersResolver;
+import org.kaaproject.kaa.client.event.FetchEventListeners;
+import org.kaaproject.kaa.client.event.registration.EndpointRegistrationManager;
+import org.kaaproject.kaa.client.event.registration.UserAttachCallback;
+import org.kaaproject.kaa.client.transact.TransactionId;
+import org.kaaproject.kaa.common.endpoint.gen.SyncResponseResultType;
+import org.kaaproject.kaa.common.endpoint.gen.UserAttachResponse;
+import org.kaaproject.kaa.schema.sample.event.thermo.ChangeDegreeRequest;
+import org.kaaproject.kaa.schema.sample.event.thermo.ThermostatEventClassFamily;
+import org.kaaproject.kaa.schema.sample.event.thermo.ThermostatEventClassFamily.DefaultEventFamilyListener;
+import org.kaaproject.kaa.schema.sample.event.thermo.ThermostatInfoRequest;
+import org.kaaproject.kaa.schema.sample.event.thermo.ThermostatInfoResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class demonstrates how to send/receive events to/from endpoints via Kaa event subsystem.
+ */
+public class EventDemo {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EventDemo.class);
+
+    // Credentials for attaching user
+    private static final String USER_EXTERNAL_ID = "userExternalId";
+    private static final String USER_ACCESS_TOKEN = "userAccessToken";
+    // Kaa client
+    private static KaaClient kaaClient;
+
+    public static void main(String[] args) {
+        LOG.info("Event demo started");
+
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e1) {
+            LOG.info("{}", e1);
+        }
+
+        kaaClient = Kaa.newClient(new DesktopKaaPlatformContext(), new SimpleKaaClientStateListener());
+        kaaClient.start();
+
+        EndpointRegistrationManager registrationManager = kaaClient.getEndpointRegistrationManager();
```

Comment:
This is strange. there should be direct method kaaClient.attachUser. Use it instead